### PR TITLE
chore: [CO-578] remove DNS/MX record checks

### PR DESCRIPTION
**Pull request by aheeva-yuliya**
DNS/MX removal

1. CheckDomainMXRecord request, response, handler, right.

2. attributes: 

- zimbraMtaDnsLookupsEnabled (deprecated)
- zimbraAdminConsoleDNSCheckEnabled (enable MX check feature for domain)
- zimbraDNSCheckHostname (DNS check by customers that configure their MX to point at spam relays or other non-zimbra inbox smtp servers)
- zimbraReverseProxyDnsLookupInServerEnabled (control whether force the server side do the DNS lookup and send the result IP back to proxy)
- zimbraDNSMasterIP (IP Address(es) of the root DNS servers to be used by the DNS cache service)
- zimbraDNSUseTCP (for dns cache, whether or not to use TCP)
- zimbraDNSUseUDP (for dns cache, whether or not to use UDP)
- zimbraDNSTCPUpstream (for dns cache, whether or not to only use TCP when talking to the upstream Master DNS servers)

 and related rights.	
	
3. Unbound DNS resolver config.

Related PRs:
https://bitbucket.org/zextras/carbonio-thirds/pull-requests/20
https://github.com/Zextras/carbonio-build/pull/53

### CODE REVIEW

**Comment by keshavbhatt** [comment]
> * zimbraMtaSmtpDnsSupportLevel (value for postconf smtp_dns_support_level)

Removing this attribute will also impact some other places, zmconfigd may fail check if the we also require to perform cleanup in `carbonio-ldap-utilities/conf/zmconfigd.cf` look for [smtp_dns_support_level](https://www.postfix.org/postconf.5.html#smtp_dns_support_level) postconf attribute.

**Comment by keshavbhatt** [comment]
> * zimbraMtaSmtpTlsDaneInsecureMXPolicy  (value for postconf smtp_tls_dane_insecure_mx_policy)
>   and related rights.

Same for this one. Check if we require cleanup in `carbonio-ldap-utilities/conf/zmconfigd.cf`

**Comment by keshavbhatt** [comment]
Check if we still require to provide and maintain `carbonio-unbound` package in carbonio-thrid repository. It seems related and required by this feature.

File: packages/common-appserver-docs/PKGBUILD [hunk 1]
```diff
@@ -98,8 +98,6 @@ package() {
     "${pkgdir}/opt/zextras/documentation/pop-imap.txt"
   install -D "store/docs/postfix-ldap-tables.txt" \
     "${pkgdir}/opt/zextras/documentation/postfix-ldap-tables.txt"
-  install -D "store/docs/postfix-split-domain.md" \
-    "${pkgdir}/opt/zextras/documentation/postfix-split-domain.md"
```
[the remark this answers is not in the document]
Comment (aheeva-yuliya) [reply packages/common-appserver-docs/PKGBUILD:102]: I can keep it if you think is usefull 
but it's not gonna be up-to-date anymore

[the remark this answers is not in the document]
Comment (frisonisland) [reply packages/common-appserver-docs/PKGBUILD:102]: We should discuss about this, but all these docs seems outdated.

File: packages/common-appserver-conf/PKGBUILD [hunk 2]
```diff
@@ -40,8 +40,6 @@ package() {
     "${pkgdir}/opt/zextras/conf/log4j.properties.in"
   install -D "store-conf/conf/stats.conf.in" \
     "${pkgdir}/opt/zextras/conf/stats.conf.in"
-  install -D "store/conf/unbound.conf.in" \
-    "${pkgdir}/opt/zextras/conf/unbound.conf.in"
```
Comment (frisonisland) [inline packages/common-appserver-conf/PKGBUILD:44]: What about this? Why remove it? For what is it needed?

**Review by frisonisland — CHANGES_REQUESTED** [review]
Everything looks ok to me except for unbound.conf.in: it is not clear why we have to remove it

File: packages/common-appserver-conf/PKGBUILD (hunk 2 above)
Comment (aheeva-yuliya) [reply packages/common-appserver-conf/PKGBUILD:44]: unbound.conf is used to configure Unbound DNS resolver
Luca asked some people to confirm about attributes i want to remove and probably I want to ask about this (Unbound product and its conf) Tell me if you know it's used or how to check if it's used

Comment (frisonisland) [reply packages/common-appserver-conf/PKGBUILD:44]: If it is not within the boundaries of the PR I would remove it another time. Does it have anything to do with check DNS and MX record?

Comment (aheeva-yuliya) [reply packages/common-appserver-conf/PKGBUILD:44]: @frisonisland @keshavbhatt I've asked Luca about Unbound. He said as far as he knows it is not used. Waiting for confirmation from Lorenzo Fascì Omar Verbi Laurent Francoise about attributes and Unbound if we still have to maintain it.

File: packages/common-appserver-docs/PKGBUILD (hunk 1 above)
[the remark this answers is not in the document]
Comment (aheeva-yuliya) [reply packages/common-appserver-docs/PKGBUILD]: @keshavbhatt keeping the doc

File: packages/common-appserver-conf/PKGBUILD (hunk 2 above)
Comment (aheeva-yuliya) [reply packages/common-appserver-conf/PKGBUILD:44]: @frisonisland removing the config

**Comment by aheeva-yuliya** [comment]
I also have this https://github.com/Zextras/carbonio-ldap-utilities/pull/35 but as far as I understood there is no need to open it, right

**Comment by keshavbhatt** [comment]
> I also have this [Zextras/carbonio-ldap-utilities#35](https://github.com/Zextras/carbonio-ldap-utilities/pull/35) but as far as I understood there is no need to open it, right

should not cause any effect but lets have it to completely get rid of these entries from our codebase. The existance of those files there is already a question as they seems not involved in build process.

File: packages/common-appserver-docs/PKGBUILD (hunk 1 above)
[the remark this answers is not in the document]
Comment (keshavbhatt) [reply packages/common-appserver-docs/PKGBUILD]: > but all these docs seems outdated

Not **All** :)

**Review by keshavbhatt — APPROVED** [review]
LGTM, 
Just make sure the new build is tested for integration.

**Review by Polpetta — APPROVED** [review]

**Pull request merged**